synapse/rest/client/v2_alpha/account.py

```diff
@@ -441,6 +455,13 @@ async def on_POST(self, request):
         existing_user_id = await self.store.get_user_id_by_threepid("msisdn", msisdn)
 
         if existing_user_id is not None:
+            if self.hs.config.request_token_inhibit_3pid_errors:
+                # Make the client think the operation succeeded but don't actually send
+                # anything. This is a compromise between sending an email, which could
+                # be a spam vector, and letting the client know which email address is
+                # bound to an account and which one isn't.
+                return 200, {"sid": random_string(16)}
```

Comment:
This is called enough times (with this same comment) that it makes me think it should be centralized somewhere.

Reply:
Not sure where though, it's very specific to a type of endpoints which is split between 5 different servlets in two different files, so there's no obvious place to put it. Plus a bunch of these servlets do very similar (if not almost identical) stuff so although I agree that it should be centralised somewhere, that's the case for most of these functions and that should probably happen in a different PR.

Reply:
Right. It was mostly the huge comment that was concerning to me. I wonder if we could make them a bit simply and refer to the comments about `request_token_inhibit_3pid_errors`:

> Make the client think the operation succeeded. See the rationale in the comments for request_token_inhibit_3pid_errors.

Or something like that? 🤷‍♂️  I think it is OK either way.

Reply:
wfm, done in https://github.com/matrix-org/synapse/pull/7315/commits/d30eab4df01c082157a60297aa1b4a6376c36720